src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/Msmq/MsmqCommon.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using System.Text;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Tagging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Msmq
+{
+    internal class MsmqCommon
+    {
+        internal static readonly IntegrationInfo IntegrationId = IntegrationRegistry.GetIntegrationInfo(MsmqConstants.IntegrationName);
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(MsmqCommon));
+
+        internal static Scope CreateScope<TMessageQueue>(Tracer tracer, string command, string spanKind, TMessageQueue messageQueue, bool? messagePartofTransaction = null)
+            where TMessageQueue : IMessageQueue
+        {
+            if (!tracer.Settings.IsIntegrationEnabled(IntegrationId))
+            {
+                // integration disabled, don't create a scope, skip this trace
+                return null;
+            }
+
+            Scope scope = null;
+
+            try
+            {
+                var tags = new MsmqTags(spanKind)
+                {
+                    Command = command,
+                    IsTransactionalQueue = messageQueue.Transactional.ToString(),
+                    Path = messageQueue.Path,
```

Comment:
For the Path tag (not the resource name), I think it would be desirable to include the machine name, that way users can filter further on that. Is that reasonable @lucaspimentel or should we not include the machine name on this tag?

Reply:
The machine name property doesn't give me the machine name but just a "."  so does seem useless..
The only way to get it is to use the format name property of the queue. So path and format name look almost the same.. 
Path: Private$\myQueue3
FormatName: DIRECT=OS:desktop-duevvft\Private$\myQueue3
we need the path for grouping by resource name: 
             `  span.ResourceName = $"{command} {messageQueue.Path}";`

I can try to isolate the machine name by removing the "Path" part from the format name, it should cover all sorts of cases?

Reply:
Got it. With that info, then I would recommend keeping what you have now so we don't spend extra cycles splitting and allocating new strings 😄